### PR TITLE
[NEW] New server selector

### LINKED ISSUE #1721: [NEW] New "Servers" view
screen shot 2018-05-25 at 14 37 18

### PR DESCRIPTION
**Pull request by rafaelks**
@RocketChat/ios 

Closes #1721

## Progress

- [x] Let user choose server from the new view;
- [x] Remove all the old views/controllers;
- [x] Remove all the old views/controllers;
- [x] Let user remove a server by sliding into the cell;
- [x] Animations;
- [x] Separate content in a view model;
- [x] Tests;

### CODE REVIEW

File: Rocket.Chat/Views/Subscriptions/ServersListView.swift [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    // MARK: Showing the View
+
+    static func showIn(_ view: UIView) -> ServersListView? {
+        guard let instance = ServersListView.instantiateFromNib() else { return nil }
+        instance.backgroundColor = UIColor.black.withAlphaComponent(0)
+        instance.headerView.alpha = 0
+        instance.frame = view.bounds
+        view.addSubview(instance)
+
+        UIView.animate(withDuration: 0.2) {
```
[the remark this answers is not in the document]
Comment (rafaelks) [reply Rocket.Chat/Views/Subscriptions/ServersListView.swift]: @filipealva There are two animations to open and two to close. One of each doesn't have curve (for the background color).

[the remark this answers is not in the document]
Comment (rafaelks) [reply Rocket.Chat/Views/Subscriptions/ServersListView.swift]: @filipealva Updated the animation to use curve in all of it.

**Comment by rafaelks** [comment]
@filipealva @cardoso Guys, this is ready to review, now with tests, view model and all strings translated. 👍

File: Rocket.Chat/Views/Subscriptions/ServersListView.swift [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+    @IBOutlet weak var tableViewHeighConstraint: NSLayoutConstraint! {
+        didSet {
+            tableViewHeighConstraint.constant = viewModel.viewHeight
+        }
+    }
+
+    private func animates(_ animations: @escaping VoidCompletion, completion: VoidCompletion? = nil) {
+        UIView.animate(withDuration: 0.15, delay: 0, options: UIViewAnimationOptions(rawValue: 7 << 16), animations: {
```
Comment (Sameesunkaria) [inline Rocket.Chat/Views/Subscriptions/ServersListView.swift]: 🤓

**Pull request merged**